modules/battery_manager/ajax/add_or_edit_entry.php

```diff
@@ -0,0 +1,207 @@
+<?php
```

Comment:
This will probably have to go into another pass of the module but:

1. Add and edit should be two different endpoints, since they're doing two different things
2. They should be implemented as handlers in the php directory, the ajax directory should be deprecated.